src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -159,10 +165,16 @@ function discoverController($scope, config, courier, $route, $window, Notifier,
   $scope.uiState = $state.makeStateful('uiState');
 
   function getStateDefaults() {
+    const defaultColumns = config.get('defaultColumns').slice();
+
+    const timeFieldName = $scope.indexPattern.timeFieldName;
+    if (timeFieldName) {
+      defaultColumns.unshift(timeFieldName);
```

Comment:
This brings up an interesting situation. Currently, if the user was to set the `defaultColumns` advanced setting to "_source, @timestamp (or the equivalent timestamp name)" the Discover app would display the following three columns: Time, @timestamp, _source.

The way that I've implemented this change mirrors this behavior, and even though we're by default adding the `indexPattern.timeFieldName` to the list of columns, it won't by default be listed in the advanced settings "defaultColumns" setting. Likewise, if the user were to add the "@timestamp" to the list of defaultColumns in the advanced settings, it would show up twice in the Discover App.

We could potentially not add the `timeFieldName` to the list of defaultColumns if it's already specified there? Would that make this behavior any more user friendly?